### PR TITLE
Parameterizing sparse pose adjustment method

### PR DESCRIPTION
**Pull request by ayrton04**
@bradleypowers @kayak0806 @tappan-at-git @davelkan

### CODE REVIEW

File: src/spa_solver.h [hunk 1]
```diff
@@ -54,6 +55,8 @@ class SpaSolver : public karto::ScanSolver
   karto::ScanSolver::IdPoseVector corrections;
 
   sba::SysSPA2d m_Spa;
+
+  int m_SpaMethod;
```
Comment (ayrton04) [inline src/spa_solver.h]: Not ROS compliant, but in keeping with the style in this file

File: src/slam_karto.cpp [hunk 2]
```diff
@@ -281,6 +281,25 @@ SlamKarto::SlamKarto() :
 
   // Set solver to be used in loop closure
   solver_ = new SpaSolver();
+
+  std::string spa_method_string;
+  int spa_method = SBA_SPARSE_CHOLESKY;
+
+  if(private_nh_.getParam("spa_method", spa_method_string))
+  {
+    if(spa_method_string == "dense_cholesky")
```
Comment (ayrton04) [inline src/slam_karto.cpp]: Again, not conformant to ROS standards, but follows the established style for this file.

**Comment by tappan-at-git** [comment]
Looks good. Shame about the established lack of style.

**Comment by bradleypowers** [comment]
Yeah, the existing style issues aside, I propose LGTM.

**Pull request merged**